udata_schema_gouvfr/views.py

```diff
@@ -5,21 +5,52 @@
 from udata import theme
 from udata.frontend import template_hook
 
+import requests
+
 blueprint = Blueprint('schema', __name__, template_folder='templates')
 
 
-def validata_url(resource):
+def validata_url(resource, schema_url=None):
     base = current_app.config.get('SCHEMA_GOUVFR_VALIDATA_URL')
-
-    query = urlencode({
-        "input": "url",
-        "schema_name": f"schema-datagouvfr.{resource.schema}",
-        "url": resource.url,
-    })
+    if schema_url:
+        query = urlencode({
+            'input': 'url',
+            'schema_url': schema_url,
+            'url': resource.url,
+        })
+    else:
+        query = urlencode({
+            'input': 'url',
+            'schema_name': f"schema-datagouvfr.{resource.schema['name']}",
+            'url': resource.url,
+        })
 
     return f"{base}/table-schema?{query}"
 
 
+def is_table_schema(schemas, current_schema):
+    if schemas:
+        for schema in schemas:
+            if schema['name'] == current_schema and schema['schema_type'] == 'tableschema':
+                return True
+    return False
```

Comment:
```suggestion
    return any([s['name'] == current_schema and s['schema_type'] == 'tableschema' for s in (schemas or [])])
```
(optional, a bit Perl like 😅